src/main/java/org/kiwiproject/registry/eureka/client/EurekaParser.java

```diff
@@ -18,18 +20,24 @@
 
     @SuppressWarnings("unchecked")
     public static List<EurekaInstance> parseEurekaResponse(Map<String, Object> response) {
-        var applications = ((Map<String, Object>) response.get("applications")).get("application");
+        checkArgumentNotNull(response, "Map data of Eureka response must not be null");
+
+        var applications = (Map<String, Object>) response.get("applications");
+
+        checkState(nonNull(applications), "Eureka data must contain a key 'applications' that contains a Map");
```

Comment:
_nit_: "Eureka response map must contain a key 'applications' that contains a Map<String, Object>"